code/Forms/UnsavedChangesIndicator.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace SilverStripe\Admin\Forms;
+
+use RuntimeException;
+
+use SilverStripe\Forms\DatalessField;
+
+/**
+ * A visual indicator to show whether a form has unsaved changes.
+ */
+class UnsavedChangesIndicator extends DatalessField
+{
+    /**
+     * An associative array of time periods to determine when the notice displays, and when it updates to a warning.
+     * notice: The number of minutes it takes to display the notice after starting to edit a form
+     * warning: The number of minutes it takes to update the display to a warning after displaying the notice.
+     */
+    private static array $minutes = [
+        'notice' => 5,
+        'warning' => 10,
+    ];
+
+    protected $schemaComponent = 'UnsavedChangesIndicator';
+
+    public function __construct($name, $title = null, $value = null)
+    {
+        // This field must be named 'UnsavedChangesIndicator' in order for
+        // applyTransforms.js to find and update it
+        return parent::__construct('UnsavedChangesIndicator', $title, $value);
+    }
```

Comment:
```suggestion
    public function __construct($title = null, $value = null)
    {
        // This field must be named 'UnsavedChangesIndicator' in order for
        // applyTransforms.js to find and update it
        return parent::__construct('UnsavedChangesIndicator', $title, $value);
    }
```
Why have a required `$name` parameter that we then don't use? We probably shouldn't have that parameter at all, but if having it makes tests easier or something then we can keep it but make it optional.

> This field must be named 'UnsavedChangesIndicator' in order for applyTransforms.js to find and update it

I don't think any other field has this requirement, I suspect it wouldn't be too hard to use something else to select the field instead and allow setting a variable name for the field like all the other fields - but it's debatable whether it'd be worth finding a way to do that.

I don't feel strongly about that part so I'll leave it up to you to decide if it's worth the extra time in order to keep this field's "name" treatment the same as all our other fields.

Reply:
Have removed all the constructor arguments since they're not needed